tests/unit/query_v2_expression_evaluator.cpp

```diff
@@ -0,0 +1,1786 @@
+    auto *op = storage.Create<SubscriptOperator>(list_literal, storage.Create<PrimitiveLiteral>(-2));
+    auto value = Eval(op);
+    EXPECT_EQ(value.ValueInt(), 3);
+  }
+  {
+    // Indexing with one operator being null.
+    auto *op = storage.Create<SubscriptOperator>(storage.Create<PrimitiveLiteral>(TypedValue()),
+                                                 storage.Create<PrimitiveLiteral>(-2));
+    auto value = Eval(op);
+    EXPECT_TRUE(value.IsNull());
+  }
+  {
+    // Indexing with incompatible type.
+    auto *op = storage.Create<SubscriptOperator>(list_literal, storage.Create<PrimitiveLiteral>("bla"));
+    EXPECT_THROW(Eval(op), ExpressionRuntimeException);
+  }
+}
+
+TEST_F(ExpressionEvaluatorTest, MapIndexing) {
+  auto *map_literal = storage.Create<MapLiteral>(
+      std::unordered_map<PropertyIx, Expression *>{{storage.GetPropertyIx("a"), storage.Create<PrimitiveLiteral>(1)},
+                                                   {storage.GetPropertyIx("b"), storage.Create<PrimitiveLiteral>(2)},
+                                                   {storage.GetPropertyIx("c"), storage.Create<PrimitiveLiteral>(3)}});
+  {
+    // Legal indexing.
+    auto *op = storage.Create<SubscriptOperator>(map_literal, storage.Create<PrimitiveLiteral>("b"));
+    auto value = Eval(op);
+    EXPECT_EQ(value.ValueInt(), 2);
+  }
+  {
+    // Legal indexing, non-existing key.
+    auto *op = storage.Create<SubscriptOperator>(map_literal, storage.Create<PrimitiveLiteral>("z"));
+    auto value = Eval(op);
+    EXPECT_TRUE(value.IsNull());
+  }
+  {
+    // Wrong key type.
+    auto *op = storage.Create<SubscriptOperator>(map_literal, storage.Create<PrimitiveLiteral>(42));
+    EXPECT_THROW(Eval(op), ExpressionRuntimeException);
+  }
+  {
+    // Indexing with Null.
+    auto *op = storage.Create<SubscriptOperator>(map_literal, storage.Create<PrimitiveLiteral>(TypedValue()));
+    auto value = Eval(op);
+    EXPECT_TRUE(value.IsNull());
+  }
+}
+
+using Vertex = memgraph::msgs::Vertex;
+using Edge = memgraph::msgs::Edge;
+using EdgeType = memgraph::msgs::EdgeType;
+using Value = memgraph::msgs::Value;
+using VertexId = memgraph::msgs::VertexId;
+using Label = memgraph::msgs::Label;
+
+memgraph::query::v2::accessors::VertexAccessor CreateVertex(std::vector<std::pair<PropertyId, Value>> props,
+                                                            const memgraph::msgs::ShardRequestManagerInterface *manager,
+                                                            Label label = {}) {
+  static size_t id = 0;
+  return {Vertex{VertexId{label, id++}, {label}}, std::move(props), manager};
+}
+
+memgraph::query::v2::accessors::EdgeAccessor CreateEdge(std::vector<std::pair<PropertyId, Value>> props,
+                                                        const memgraph::msgs::ShardRequestManagerInterface *manager) {
+  auto edge = Edge{.src = VertexId{{}, 0},
+                   .dst = VertexId{{}, 0},
```

Comment:
If I am not mistaken, then `0` initializes the `PrimaryKey` part of the `VertexId`. It is a bit confusing to me, usually we use `{}` to initialize a vector.